projects/output-and-error-handling/client/main.go

```diff
@@ -0,0 +1,182 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+)
+
+// Comments ----------
+// I write a lot of comments, it has helped me the entire course to learn, understand and remember
+// They can all be removed at the end to clean it up
+
+// Error Messages ----------
+// The Errors are deliberately numbered [x] so I could follow them up the call stack
+// They can be replaced with "user friendly" messages later
+
+func parseResponseBody(response *http.Response) (string, error) {
+	// check if there is a response body
+	if response.Body == nil {
+		return "", fmt.Errorf("[2] there is no response body")
+	}
+
+	// try to read the response body
+	responseBody, err := io.ReadAll(response.Body)
+
+	// if we cannot read the response body then return an error
+	if err != nil {
+		return "", fmt.Errorf("[2] failed to read response body : %w", err)
+	}
+
+	return string(responseBody), nil
+}
+
+func parseRetryAfterHeader(retryAfterHeaderString string) (time.Duration, error) {
+	// fmt.Println("--- DEBUG retryAfterHeaderString", retryAfterHeaderString)
+
+	// first condition "a while"
+	if retryAfterHeaderString == "a while" {
+		return 0, fmt.Errorf("[3] retry-after header is 'a while'")
```

Comment:
What do you think we should do if the server returns "a while"? It seems like a strange thing to special-case to give this specific error...

Reply:
i have removed this case, so if it's not a valid integer from string, or timestamp from string it defaults to 3rd case behaviour